src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/DefaultModelMetadataProvider.cs

```diff
@@ -24,6 +24,8 @@ public class DefaultModelMetadataProvider : IModelMetadataProvider
         public DefaultModelMetadataProvider(ICompositeMetadataDetailsProvider detailsProvider)
         {
             DetailsProvider = detailsProvider;
+
+            _cacheEntryFactory = CreateCacheEntry;
```

Comment:
Method to delegate conversion allocates (per the spec). That means that every call, every cache lookup, we 'new' a small object. I wouldn't always do this optimization but it did shop up as part of the profiling that led me to make this change.

There's a cool VS Extension that will bring this kind of thing to your attention https://visualstudiogallery.msdn.microsoft.com/f9b47b93-8675-4ae0-9c52-5da8027c4bb8

It will show you in the editor anywhere where the code you wrote is going to allocate without you writing 'new'. I don't recommend fixing the things it shows you 99% of the time, but it's a good educational tool